### PR TITLE
Prerelease/winter20

### PR DESCRIPTION
**Pull request by pozil**

### CODE REVIEW

**Comment by pozil** [comment]
jest tests are failing after update.
Opened an issue: https://github.com/salesforce/sfdx-lwc-jest/issues/96

**Comment by pozil** [comment]
@muenzpraeger I need your help on a CI error.

I'm getting a `navigationMenus` (Community menu) deployment error on CI:
```bash
ERROR running force:source:push:  Unexpected file found in package directory: /home/circleci/project/force-app/main/default/navigationMenus/Default_Navigation.navigationMenu-meta.xml
Exited with code 1
```

What's weird about this is that it works fine on v47 scratch orgs that I spin myself. It's part of the newly added metadata so it should work.

Is there something special about the DevHub org associated with `prerelease*` branches that could explain the difference?

**Comment by muenzpraeger** [comment]
Is it a new metadata type?

The CLI runs on 46 to catch these things. ;-)

**Comment by pozil** [comment]
Yes it's new. I thought that you had a special flow for `prerelease` branches.
How can we proceed?

**Comment by muenzpraeger** [comment]
Is it needed? Or just got automatically added? Asking because it can be standard metadata that’s not relevant for source.

Like here:
https://github.com/trailheadapps/ebikes-lwc/issues/83

**Comment by pozil** [comment]
Yes, it's pretty important, it's the community menu:
Screenshot 2019-10-09 at 21 28 43

If you don't have that, there's only "Home" in the menu by default. You then have to manually add the pages.

**Comment by muenzpraeger** [comment]
Add after [this](https://github.com/trailheadapps/ebikes-lwc/blob/master/.circleci/config.yml#L65) an install of the DX plugin for `salesforcedx@prerelease`.

https://www.npmjs.com/package/salesforcedx#install-a-tagged-version-of-salesforce-dx

And don't forget to revert after merge to `master` with GA... ;-)

**Comment by pozil** [comment]
Damn it, I wish we could have the same prerelease dist name everywhere: it's `pre-release` (with a dash) for the salesforcedx plugin.

**Comment by pozil** [comment]
@muenzpraeger I don't get it, I could use your CI knowledge here.

Prerelease CLI is now correctly installed in CI as per versions displayed in logs but I still have the same metadata deployment error.

I did a few things to double check if I made a silly mistake but I couln't find anything obvious:
1. I've compared plugin versions with my local CLI and it all matches.
1. I've re-cloned the repo and recreated several scratch orgs and it works all of the time.
1. I also re-ran the CI build but not luck.

**Comment by muenzpraeger** [comment]
I ran a quick test, for some (whatever) reason the linked plugin isn't preserved, see [here](https://circleci.com/gh/trailheadapps/ebikes-lwc/1048).

Just for verification of all the other changes temporary disabled pushing the new community metadata using the `forceignore` [here](https://github.com/trailheadapps/ebikes-lwc/pull/81/files#diff-12dbbeda74918448da11bcb672085873R23). Now, that's not super great. But given that we have one big commit with all the changes there's now verification that all the other changes are good.

After R2 drops on Saturday night the `forceignore` needs to be reverted, and CI tested then again the GA version.

**Comment by pozil** [comment]
Wow that's scary. I wonder where the plugin comes from in the deploy job...
Thanks for your help, this was the last thing that needed to be changed. We're good to go.

**Comment by muenzpraeger** [comment]
At some point ago they started to bundle it with the CLI. Post this release we should investigate if changing to installing the CLI binary (compared to the npm approach) works better.

File: .forceignore [hunk 1]
```diff
@@ -19,3 +19,6 @@ package.xml
 
 # Allowing EBikes profile for source push/pull
 !**/E-Bikes Profile.profile-meta.xml
+
+# Temporary disable Community navigation metadata
+**navigationMenus
```
Comment (muenzpraeger) [inline .forceignore]: Doesn't this need to be undone?

Comment (pozil) [reply .forceignore]: Yes, let me try. I was about to do that in a separate PR because of the prerelease stuff but I guess that it should work now.

**Comment by pozil** [comment]
@muenzpraeger fixed

**Pull request merged**